.github/workflows/do_not_merge.yml

```diff
@@ -6,15 +6,21 @@ on:
 
 jobs:
   do-not-merge:
-    if: ${{ contains(github.event.*.labels.*.name, 'DNM') ||
-            contains(github.event.*.labels.*.name, 'TSC') ||
-            contains(github.event.*.labels.*.name, 'Architecture Review') ||
-            contains(github.event.*.labels.*.name, 'dev-review') }}
     name: Prevent Merging
     runs-on: ubuntu-22.04
     steps:
-      - name: Check for label
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.11
+
+      - name: Install Python dependencies
+        run: |
+          pip install PyGithub
+
+      - name: Run the check script
         run: |
-          echo "Pull request is labeled as 'DNM', 'TSC', 'Architecture Review' or 'dev-review'."
-          echo "This workflow fails so that the pull request cannot be merged."
-          exit 1
+          ./scripts/ci/do_not_merge.py -p "${{ github.event.pull_request.number }}"
```

Comment:
Not really sure why we need to convert this simple workflow into a Python script. If we ever need to handle more complex conditions, that can be done within the workflow itself using the [`github-script` action](https://github.com/actions/github-script/).